app/models.py

```diff
@@ -0,0 +1,81 @@
+from typing import Optional
+from uuid import UUID, uuid4
+from datetime import datetime, timedelta, timezone
+from jose import JWTError, jwt
+from pydantic import BaseModel, EmailStr, Field
+from passlib.context import CryptContext
+
+#Set up password hashing context using bcrypt
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+#JWT settings
+SECRET_KEY = "your_secret_key"
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+
+class UserBase(BaseModel):
+    username: str
+    email: EmailStr
+    is_admin: bool = False
+    is_active: bool = True
+
+class UserCreate(UserBase):
+    password: str
+
+#Inheriting from UserBase
+class User(UserBase):
+    id: UUID = Field(default_factory=uuid4)
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+
+    def get_password_hash(self):
```

Comment:
The validation should be before calling the constructor

Reply:
done.